ctapipe/core/component.py

```diff
@@ -1,10 +1,45 @@
 """ Class to handle configuration for algorithms """
-
 from traitlets.config import Configurable
 from traitlets import TraitError
 from abc import ABCMeta
 from logging import getLogger
 
+from inspect import isabstract
+
+
+def non_abstract_children(base):
+    """
+    Return all non-abstract subclasses of a base class recursively.
+    """
+    subclasses = base.__subclasses__() + [
+        g for s in base.__subclasses__()
+        for g in non_abstract_children(s)
+    ]
+    non_abstract = [g for g in subclasses if not isabstract(g)]
+
+    return non_abstract
+
+
+def subclass_from_name(base, name=None, *args, **kwargs):
```

Comment:
add `@lru_cache`?

Reply:
I would refrain from optimizations at this point. I think, the usual use case for this function is once in the lifetime of a tool at the beginning, so not much to cache here. But we can certainly do this when we have the impression this is a time-sink.

Reply:
Also if you cache this, then it might not detect new children that have been added to the global space since the previous call

Reply:
So we resolve this comment